src/t_hash.c

```diff
@@ -551,21 +558,14 @@ SetExRes hashTypeSetExpiryListpack(HashTypeSetEx *ex, sds field,
     if (unlikely(checkAlreadyExpired(expireAt))) {
         propagateHashFieldDeletion(ex->db, ex->key->ptr, field, sdslen(field));
         hashTypeDelete(ex->hashObj, field, 1);
-        
-        /* get listpack length */
-        listpackEx *lpt = ((listpackEx *) ex->hashObj->ptr);
-        unsigned long length = lpLength(lpt->lp) / 3;
-        updateKeysizesHist(ex->db, getKeySlot(ex->key->ptr), OBJ_HASH, length+1, length); 
```

Comment:
Please pay attention that I moved keysize update to the command handler function. Rather than doing updates on different places and try to be accurate, we may do sth like: 

```
void hgetexCommand() {
  oldlen = hashTypeLen(o, 0);
  
  ....
  handle command
  ...
  
  newlen = hashTypeLen(o, 0);
  if (oldlen != newlen)
     updateKeysizesHist(...)
}
```